.circleci/setup-opam

```diff
@@ -0,0 +1,16 @@
+#! /usr/bin/env bash
+#
+# Set up opam and install opam packages required to build and test atd
+#
+# This is used by the Dockerfile and by the CI build.
+#
+set -eu
+
+eval "$(opam env)"
+
+opam repo add --all-switches -k git \
+  github https://github.com/ocaml/opam-repository.git
+opam repo remove default
+# no need for 'opam update'
```

Comment:
is it faster than doing just `opam update` or `opam update -a`? Seems a bit overkill to clone the whole repo every time

Reply:
The Docker image comes configured only with a local repository of opam packages (`~/opam-repository`) rather than the normal opam-repository on GitHub. I'm not sure why.

```
$ docker run -it ocaml/opam:ubuntu-20.04-ocaml-4.13
opam@6a6e0c37bd18:~$ ls
opam-repository
opam@6a6e0c37bd18:~$ date       
Thu Feb 24 06:08:30 UTC 2022
opam@6a6e0c37bd18:~$ ls -l
total 4
drwxr-xr-x 5 opam opam 4096 Feb 19 23:17 opam-repository
```
shows that the local repo is 5 days old, which is great, but I don't know if it's just luck or if it gets refreshed often enough. https://hub.docker.com/r/ocaml/opam is the home of those Docker images but I don't see the info about the freshness we can expect from the pre-downloaded opam-repository.